src/main/java/de/uhd/ifi/se/decision/management/jira/extraction/impl/GitClientImpl.java

```diff
@@ -2,940 +2,934 @@
+	}
+
+	private boolean openRepository(String repoUri, File directory) {
+		try {
+			Git git = Git.open(directory);
+			gits.put(repoUri, git);
+		} catch (IOException e) {
+			LOGGER.error(
+					"Git repository could not be opened: " + directory.getAbsolutePath() + "\n\t" + e.getMessage());
+			return false;
+		}
 		return true;
-	    }
-	    return false;
-	}
-    }
-
-    private boolean isRepoOutdated(long lastModified) {
-	Date date = new Date();
-	long fileLifespan = date.getTime() - lastModified;
-	return fileLifespan > REPO_OUTDATED_AFTER;
-    }
-
-    private boolean updateFileModifyTime(File file) {
-	Date date = new Date();
-	if (!file.setLastModified(date.getTime())) {
-	    LOGGER.error("Could not modify a file modify time, repositories will be fetched on each request.");
-	    return false;
-	}
-	return true;
-    }
-
-    private boolean cloneRepository(String uri, File directory) {
-	if (uri == null || uri.isEmpty()) {
-	    return false;
-	}
-	try {
-	    Git git = Git.cloneRepository().setURI(uri).setDirectory(directory).setCloneAllBranches(true).call();
-	    gits.put(uri, git);
-	    setConfig(uri);
-	} catch (GitAPIException e) {
-	    LOGGER.error("Git repository could not be cloned: " + uri + " " + directory.getAbsolutePath() + "\n\t"
-		    + e.getMessage());
-	    return false;
-	}
-	// TODO checkoutDefault branch
-	return true;
-    }
-
-    private boolean initRepository(String repoUri, File directory) {
-	try {
-	    Git git = Git.init().setDirectory(directory).call();
-	    gits.put(repoUri, git);
-	} catch (IllegalStateException | GitAPIException e) {
-	    LOGGER.error("Bare git repository could not be initiated: " + directory.getAbsolutePath());
-	    return false;
-	}
-	return true;
-    }
-
-    private boolean setConfig(String repoUri) {
-	Repository repository = this.getRepository(repoUri);
-	StoredConfig config = repository.getConfig();
+	}
+
+	private boolean pull(String repoUri) {
+		LOGGER.info("Pulling Repository: " + repoUri);
+		if (!isPullNeeded(repoUri)) {
+			try {
+				System.out.println("Unneccesary: " + getRepository(repoUri).getBranch());
+			} catch (IOException e) {
+				// TODO Auto-generated catch block
+				e.printStackTrace();
+			}
+			return true;
+		}
+		try {
+			System.out.println("Is necessary: " + repoUri);
+			ObjectId oldHead = getRepository(repoUri).resolve("HEAD^{tree}");
+			System.out.println("oldHead: " + oldHead);
+			List<RemoteConfig> remotes = gits.get(repoUri).remoteList().call();
+			for (RemoteConfig remote : remotes) {
+				gits.get(repoUri).fetch().setRemote(remote.getName()).setRefSpecs(remote.getFetchRefSpecs())
+						.setRemoveDeletedRefs(true).call();
+				LOGGER.info("Fetched branches in " + gits.get(repoUri).getRepository().getDirectory());
+			}
+			gits.get(repoUri).pull().call();
+
+
+			System.out.println("Pulled ");
+			ObjectId head = getRepository(repoUri).resolve("HEAD^{tree}");
+			System.out.println("NewHead: " + head);
+			if (oldHead != head) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use equals() to compare object references.](https://app.codacy.com/manual/UHD/cures-condec-jira/pullRequest?prid=5229259)